pkg/scheduler/eventhandlers.go

```diff
@@ -160,6 +160,16 @@ func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
 
 	logger.V(4).Info("Update event for unscheduled pod", "pod", klog.KObj(newPod))
 	sched.SchedulingQueue.Update(logger, oldPod, newPod)
+	if hasNominatedNodeNameChanged(oldPod, newPod) {
+		// Nominated node changed in pod, so we need to treat it as if the pod was deleted,
+		// because the lower or equal priority pods treat such a pod as if it was assigned.
```

Comment:
```suggestion
		// Nominated node changed in pod, so we need to treat it as if the pod was deleted from the old nominated node,
		// because the scheduler treats such a pod as if it was already assigned when scheduling lower or equal priority pods.
```